internal/monitor/monitor.go

```diff
@@ -151,7 +152,37 @@ func Unlock(podkey string) {
 	PodMonitor.PodKeyMap.Delete(podkey)
 }
 
-// Watch watches a watcher result channel. Upon receiving an event, the fn WatchFunc is invoked.
+// StoreNodeUID store node name and UID
+func (pm *PodMonitorType) StoreNodeUID(nodeName, uid string) {
+	log.Infof("StoreNodeUid added node: %s uid: %s", nodeName, uid)
+	pm.NodeNameToUID.Store(nodeName, uid)
+}
+
+// GetNodeUID returns the node UID
+func (pm *PodMonitorType) GetNodeUID(nodeName string) string {
+	log.Infof("GetNodeUid added node: %s", nodeName)
+	any, ok := pm.NodeNameToUID.Load(nodeName)
+	if !ok {
+		return ""
+	}
+	uid := fmt.Sprintf("%v", any)
+	return uid
+}
+
+// ClearNodeUID remove the node name UID on node
+func (pm *PodMonitorType) ClearNodeUID(nodeName, oldNodeUID string) bool {
+	// Replace with CompareAndSwap in go version 1.20
+	oldUID, ok := pm.NodeNameToUID.Load(nodeName)
+	old := fmt.Sprintf("%v", oldUID)
+	if ok && string(old) == oldNodeUID {
+		pm.NodeNameToUID.Store(nodeName, "")
+		return true
+	}
+	return false
+	// return pm.NodeNameToUid.CompareAndSwap(oldNodeUid, "")
```

Comment:
Should this comment removed?

Reply:
For some reason CompareAndSwap wasn't available in vscode, it is available now, updated to use that func